src/test/java/org/jenkinsci/plugins/workflow/cps/DeserializationTest.java

```diff
@@ -0,0 +1,136 @@
+package org.jenkinsci.plugins.workflow.cps;
+
+import groovy.lang.GroovyShell;
+import hudson.model.Result;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+import javax.annotation.CheckForNull;
+
+public class DeserializationTest {
```

Comment:
Since the new tests are related to restarts and the load step, I'd move them to `RestartingLoadStepTest` instead of making a new file (even though `existingVariablesOnRestart` does not use `load`, it's exercising the logic added for `load` steps, so I think it would be fine there). You can pass multiple method names to `@TestExtension`, like `@TestExtension(value={"methodA", "methodB"})` so that `InjectedVariable` only activates for the new tests.

Reply:
Did you mean to delete this file? It looks like you moved the tests over to `RestartingLoadStepTest`.